### PR TITLE
DB WAL size limit

### PR DESCRIPTION
**Pull request by arkpar**
Set WAL size limit to 64mb
Also includes DB detection fix made in the beta branch.

### CODE REVIEW

File: parity/params.rs [hunk 1]
```diff
@@ -100,20 +99,13 @@ impl Pruning {
...
-				Ok(db) => db,
-				Err(_) => return 0,
-			};
-			let db = journaldb::new(Arc::new(db), *i, client::DB_COL_STATE);
-			trace!(target: "parity", "Looking for best DB: {} at {:?}", i, db.latest_era());
-			db.latest_era().unwrap_or(0)
+			let mut client_path = dirs.client_path(genesis_hash, fork_name, *i);
+			client_path.push("CURRENT");
+			fs::metadata(&client_path).and_then(|m| m.modified()).ok()
 		}).unwrap()
```
Comment (tomusdrw) [inline parity/params.rs]: Will fail if all paths result in `Err -> None` should probably be `unwrap_or_default`

File: parity/params.rs [hunk 2]
```diff
@@ -100,20 +99,13 @@ impl Pruning {
 
 	fn find_best_db(dirs: &Directories, genesis_hash: H256, fork_name: Option<&String>) -> Algorithm {
 		let mut algo_types = Algorithm::all_types();
-
-		// if all dbs have the same latest era, the last element is the default one
+		// if all dbs have the same modification time, the last element is the default one
```
Comment (tomusdrw) [inline parity/params.rs]: Comment is a bit misleading - if all `fs` calls return Errors, it will panic.

Comment (arkpar) [inline parity/params.rs]: How? algo_types always has at least one element

File: parity/params.rs (hunk 1 above)
Comment (arkpar) [inline parity/params.rs]: `algo_types` can't be empty so this can't panic

Comment (tomusdrw) [inline parity/params.rs]: Right, sorry. I thought that there is `filter` being done there.

File: parity/params.rs (hunk 2 above)
Comment (tomusdrw) [inline parity/params.rs]: Right, my bad.

**Comment by rphmeier** [comment]
isn't WAL meant to be a failsafe in case of bad shutdown? Does limiting the size preserve this behavior?

**Comment by arkpar** [comment]
64MB should be enough for a failsafe. I've tested for unexpected shutdowns on a linux VM and it appears to be working fine.

**Pull request merged**